src/Api/Endpoints/Magic.cs

```diff
@@ -0,0 +1,31 @@
+using Passwordless.Api.Authorization;
+using Passwordless.Common.Constants;
+using Passwordless.Service.Features;
+using Passwordless.Service.MagicLinks;
+using Passwordless.Service.MagicLinks.Models;
+using static Microsoft.AspNetCore.Http.Results;
+
+namespace Passwordless.Api.Endpoints;
+
+public static class MagicEndpoints
+{
+    public static void MapMagicEndpoints(this IEndpointRouteBuilder app)
+    {
+        app.MapPost("/magic-link/send", async (
+                SendMagicLinkRequest request,
+                IFeatureContextProvider provider,
+                MagicLinkService magicLinkService
+            ) =>
+            {
+                // check if generate setting is on
+                if (!(await provider.UseContext()).IsMagicLinksEnabled) return Forbid();
```

Comment:
We want to make this error message helpful and guide the developer to fix the problem.